### PR TITLE
feat(devbox): devbox basic frontend ui and logic

### PR DESCRIPTION
**Pull request by mlhiter**

### CODE REVIEW

**Comment by mlhiter** [comment]
@zjy365 review request :)

File: frontend/providers/devbox/app/api/getDevboxList/route.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          port: item.spec.rules[0].http.paths[0].backend.service.port.number,
+          protocol: item.metadata.annotations['nginx.ingress.kubernetes.io/backend-protocol'],
+          openPublicDomain: !!item.metadata.labels[publicDomainKey],
+          publicDomain: item.metadata.labels[publicDomainKey],
+          customDomain: customDomain || ''
+        }
+      })
+      if (item.spec.network.extraPorts.length !== 0) {
+        const { body: service } = await k8sCore.readNamespacedService(devboxName, namespace)
+        item.networks = item.spec.network.extraPorts.map(async (network: any) => {
```
Comment (zjy365) [inline frontend/providers/devbox/app/api/getDevboxList/route.ts:79]: item.networks 这里调整一下，容易混淆

File: frontend/providers/devbox/app/api/platform/getNamespace/route.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { NextRequest } from 'next/server'
```
Comment (zjy365) [inline frontend/providers/devbox/app/api/platform/getNamespace/route.ts:1]: namespace 可以直接从 kc 里面读

File: frontend/providers/devbox/components/DevboxStatusTag.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  showBorder = false
+}: {
+  status: DevboxStatusMapType
+  showBorder?: boolean
+}) => {
+  const label = status.label
+  const t = useTranslations()
+
+  return (
+    <>
```
Comment (zjy365) [inline frontend/providers/devbox/components/DevboxStatusTag.tsx:18]: 删掉就可以

File: frontend/providers/devbox/deploy/manifests/appcr.yaml.tmpl [hunk 4]
```diff
@@ -0,0 +1,18 @@
+apiVersion: app.sealos.io/v1
+kind: App
+metadata:
+  name: devbox
+  namespace: app-system
+spec:
+  data:
+    desc: Devbox
+    url: 'https://devbox.dev.sealos.plus'
```
Comment (zjy365) [inline frontend/providers/devbox/deploy/manifests/appcr.yaml.tmpl:9]: deploy 的域名这些，要写成模板的格式

File: frontend/providers/devbox/deploy/scripts/update-backup-label.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
```
Comment (zjy365) [inline frontend/providers/devbox/deploy/scripts/update-backup-label.sh:2]: 这个文件没有使用的话，就删除吧

File: frontend/providers/devbox/stores/devbox.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        })
+        return res
+      },
+      devboxDetail: {} as DevboxDetailType,
+      setDevboxDetail: async (devboxName: string) => {
+        const res = await getMyDevboxList()
+
+        const detail = res.find((item) => item.name === devboxName) as DevboxDetailType
+
+        console.log(detail)
```
Comment (zjy365) [inline frontend/providers/devbox/stores/devbox.ts:53]: 日志删掉吧

File: frontend/providers/devbox/utils/cookie.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import Cookies from 'js-cookie'
```
Comment (zjy365) [inline frontend/providers/devbox/utils/cookie.ts:1]: next-intl i18n，应该也不需要这个文件了

**Review by zjy365 — APPROVED** [review]

**Review by zjy365 — APPROVED** [review]

**Comment by mlhiter** [comment]
@zzjin ask for merge.

File: frontend/packages/ui/src/components/icons/ExpandMoreIcon.tsx [hunk 8]
```diff
@@ -1,3 +1,5 @@
+'use client';
+'use client';
```
Comment (zzjin) [inline frontend/packages/ui/src/components/icons/ExpandMoreIcon.tsx:2]: DUP.

File: frontend/providers/devbox/services/monitorFetch.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  const { url, params } = props
+  const queryString = typeof params === 'object' ? new URLSearchParams(params).toString() : params
+  const requestOptions = {
+    method: 'GET',
+    headers: {
+      Authorization: encodeURIComponent(kubeconfig)
+    }
+  }
+
+  const doMain = process.env.MONITOR_URL || 'http://launchpad-monitor.sealos.svc.cluster.local:8428'
```
Comment (zzjin) [inline frontend/providers/devbox/services/monitorFetch.ts:13]: Hard-coded SVC url?

File: frontend/packages/ui/src/components/icons/ExpandMoreIcon.tsx (hunk 8 above)
Comment (mlhiter) [reply frontend/packages/ui/src/components/icons/ExpandMoreIcon.tsx:2]: fixed.

File: frontend/providers/devbox/services/error.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
+export const ERROR_TEXT: Record<string, string> = {
+  ETIMEDOUT: '服务器超时'
```
[the remark this answers is not in the document]
Comment (mlhiter) [reply frontend/providers/devbox/services/error.ts:2]: Copy from dbprovider,so it's not processed.
This part is in server side,i18n is not easy.
But I can transform them to english.

File: frontend/providers/devbox/services/monitorFetch.ts (hunk 9 above)
Comment (mlhiter) [reply frontend/providers/devbox/services/monitorFetch.ts:13]: The second part is the case that cannot be reached, I will delete it.

**Review by zzjin — APPROVED** [review]

**Pull request merged**